src/config/mod.rs

```diff
@@ -650,41 +699,39 @@ impl CompositeDeviceConfig {
             device
         );
 
-        let evdev_config = evdev_config.clone();
-
-        if let Some(name) = evdev_config.name {
+        if let Some(name) = evdev_config.name.as_ref() {
```

Comment:
can also be done as
```rust
        if let Some(name) = &evdev_config.name {
```
which is more concise. Keep in mind that this change isn't required at all, since `String` is clone and not copy, the string is moved to here and ownership taken, then the `.as_str()` is called below to get a str ref.

Personally I would drop these changes from the commit to reduce noise.